### DISCUSSION TITLE
PrimeNG unstyled mode and templates

### DISCUSSION THREAD

**Discussion by KasGeDev** [primefaces/community#200] [PrimeNG Templates] [answered]
Currently, we are in the decision-making process regarding whether to switch our Angular application to [PrimeNG](https://primeng.org) in the future. We find your upcoming release, especially the unstyled mode, very interesting. Will this unstyled mode also work with the [Diamond](https://www.primefaces.org/diamond-ng) template?

> Note: In upcoming versions, theming architecture will be redesigned to utilize CSS variables instead of SCSS variables in a backward compatible way for a dynamic approach. In addition, a new Unstyled mode will be provided as an alternative to the default styling so that CSS libraries like Tailwind or Bootstrap can be used to style the components. This work is planned to be completed in Q3 2023.

**Comment by cagataycivici** [comment, marked as answer]
Thank you for your interested. Unstyled mode is done for PrimeVue and PrimeReact now we're building new libraries e.g. with Tailwind;

Here is a video [preview](https://youtu.be/88qaX2dF_ik?si=K5a1YeAdkrToA3Bo).

PrimeNG is the next one to get these, first we need to implement pass through props and then bring in the unstyled mode. Our estimate is early next year for the availability.

Diamond is a premium template, it is built on top of the styled mode of PrimeNG. Due to backward compatibility concerns, we'll keep it as it is and offer new templates based on CSS libraries like Tailwind where Prime UI libraries work in unstyled mode with the seamless integration to the layout. For example, with this approach if you switch to dark modes or change primary color css variables both the layout and components can share the same design. Note that these are already available in Diamond or other templates based on SaaS. New unstyled mode allows implementing the same with CSS library of your choice like Tailwind.

**Reply by colfin-96** [reply]
Hey there,

we're really excited for the PrimeNG Unstyled Mode. Do you have a current status or overview of what is left to do? Is "early next year" still realistic? I ask as we are basing some decisions on this so it would help us with planning.

Thanks :)

**Reply by gokmensercan** [reply]
I'm writing to inquire about the current status of Unstyled Mode for PrimeNG. Back in October 2023, you mentioned that it was estimated to be ready 'early next year.' Could you please provide an update on the progress made since then? We're basing some decisions on this information and having a clearer timeline would greatly assist us in our planning.

Thank you for your time and assistance.

**Reply by joshua-kissinger** [reply]
I would also be interested in the current status of this work. It's now essentially the middle of 2024; is there any update on a release timeline?

**Reply by stewieoO** [reply]
Almost a year later still waiting for any update unfortunately.
@cagataycivici Can you tell us anything about it?
